src/lib/dgraph.rkt

```diff
@@ -0,0 +1,166 @@
+#lang typed/racket/base
+
+(require "dhash.rkt")
```

Comment:
The graph library isn't typed to my knowledge, and I didn't think it would guarantee determinism. The "d" in `dgraph` stands for deterministic. Which makes it a lot easier to write unit tests for. I would be happy to convert to using the graph library after we escape typed-racket, but I think that we are still going to need to implement deterministic graphs for testing purposes. We could hide these behind the generic interface that the library provides.